subsys/net/ip/connection.c

```diff
@@ -174,67 +175,75 @@ static struct net_conn *conn_find_handler(uint16_t proto, uint8_t family,
 			continue;
 		}
 
-		if (remote_addr) {
```

Comment:
The changes here were a bit difficult to read. Are we just swapping the remote/local address check branches here, if true, why do we need to do it.